plugins/dominant-color-images/hooks.php

```diff
@@ -186,3 +186,79 @@ function dominant_color_render_generator(): void {
 	echo '<meta name="generator" content="dominant-color-images ' . esc_attr( DOMINANT_COLOR_IMAGES_VERSION ) . '">' . "\n";
 }
 add_action( 'wp_head', 'dominant_color_render_generator' );
+
+/**
+ * Adds inline CSS for dominant color styling in the WordPress admin area.
+ *
+ * This function registers and enqueues a custom style handle, then adds inline CSS
+ * to apply background color based on the dominant color for attachment previews
+ * in the WordPress admin interface.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_admin_inline_style(): void {
+	$handle = 'dominant-color-admin-styles';
+	// PHPCS ignore reason: Version not used since this handle is only registered for adding an inline style.
+	// phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = '.wp-core-ui .attachment-preview[data-dominant-color]:not(.has-transparency) { background-color: var(--dominant-color); }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+add_action( 'admin_enqueue_scripts', 'dominant_color_admin_inline_style' );
+
+/**
+ * Adds a script to the admin footer to modify the attachment template.
+ *
+ * This function injects a JavaScript snippet into the admin footer that modifies
+ * the attachment template. It adds attributes for dominant color and transparency
+ * to the template, allowing these properties to be displayed in the media library.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_admin_script(): void {
+	?>
+	<script>
+		(function() {
+			var s = jQuery( '#tmpl-attachment' )
+			var transparencyClassName = 'true' === data.hasTransparency ? 'has-transparency' : 'not-transparent';
+			var n = s[0].innerText.replace( '{{ data.orientation }}"', '{{ data.orientation }} {{ transparencyClassName }}"	data-dominant-color="{{ data.dominantColor }}" data-has-transparency="{{ data.hasTransparency }}" style="--dominant-color: #{{ data.dominantColor }};"')
+			s.replaceWith( '<script type="text/html" id="tmpl-attachment">' + n  );
+		}());
+	</script>
+	<?php
+}
+add_action( 'admin_print_footer_scripts', 'dominant_color_admin_script', 1000 );
```

Comment:
Incorporating my above feedback, I think this can be further simplified:

1. Use a module so that we don't worry about needing the IIFE to eliminate global scope pollution.
2. Use a module so that we're guaranteed that the script only executes once the DOM has fully loaded, so no need for specifying a 1000 priority.
3. Eliminate the use of jQuery in favor of using plain JS calls.
4. Use `textContent` instead of `innerText` since it seems to be the better choice here.
5. Update the text content instead of replacing the entire element.

```suggestion
	<script type="module">
		const tmpl = document.getElementById( 'tmpl-attachment' );
		if ( tmpl ) {
			tmpl.textContent = tmpl.textContent.replace(
				'{{ data.orientation }}"',
				'{{ data.orientation }} {{ \'true\' === data.hasTransparency ? \'has-transparency\' : \'not-transparent\' }}"	data-dominant-color="{{ data.dominantColor }}" data-has-transparency="{{ data.hasTransparency }}" style="--dominant-color: #{{ data.dominantColor }};"'
			);
		}
	</script>
	<?php
}
add_action( 'admin_print_footer_scripts', 'dominant_color_admin_script' );
```

Reply:
love the new JS